tests/unit/test-util/test.adb.js

```diff
@@ -935,6 +935,41 @@ describe('utils/adb', () => {
       );
     });
 
+    it('start reference browser without APK component', async () => {
+      const adb = getFakeADBKit({
+        adbClient: {
+          startActivity: sinon.spy(() => Promise.resolve()),
+        },
+        adbkitUtil: {
+          readAll: sinon.spy(() => Promise.resolve(Buffer.from('\n'))),
+        },
+      });
+      const adbUtils = new ADBUtils({adb});
+
+      const promise = adbUtils.startFirefoxAPK(
+        'device1',
+        'org.mozilla.reference.browser',
+        undefined, // firefoxApkComponent
+        '/fake/custom/profile/path',
+      );
+
+      await assert.isFulfilled(promise);
+
+      sinon.assert.calledOnce(adb.fakeADBClient.startActivity);
+      sinon.assert.calledWithMatch(
+        adb.fakeADBClient.startActivity, 'device1', {
+          action: 'android.activity.MAIN',
+          component: 'org.mozilla.reference.browser/' +
+            'org.mozilla.reference.browser.BrowserActivity',
+          extras: [{
+            key: 'args',
+            value: '-profile /fake/custom/profile/path',
+          }],
+          wait: true,
+        }
+      );
+    });
+
```

Comment:
This test case looks pretty good!
I would just like a few more changes to also cover the opposite scenario (when a `firefoxApkComponent` has been passed explicitly as not undefined).

The new test case should be pretty similar to the one you already wrote, they will basically only differ for the `firefoxApkComponent` parameter and the resulting `component` property that we expect to get from the `startActivity`'s `calledWithMatch`, and so it should be possible to just parametrize this test case and run it twice, something like the following should allow us to cover both without duplicating almost the same test case twice:

```
async function testReferenceBrowserApkComponent(
  firefoxApkComponent, expectedApkComponent
) {
  // almost everything exactly like the test case you already wrote, with just
  // the following small changes: 
  ...
  const promise = adbUtils.startFirefoxAPK(..., firefoxApkComponent, ...);
  ...
  const apkName = 'org.mozilla.reference.browser';
  const component =
    `${apkName}/${apkName}.${expectedApkComponent}`
  sinon.assert.calledWithMatch(
    ..., {
     ...
     component,
     ...
    }); 
}

it('starts reference browser with default APK component if not specified', () => {
  return testReferenceBrowserApkComponent(undefined, 'BrowserActivity');
});

it('starts reference browser with the given APK component', () => {
  return testReferenceBrowserApkComponent('CustomActivity', 'CustomActivity');
});
```

Let me know if there is any issue with this approach that you would need help with (I haven't tried the code snippet above, and so there may be some typo or some other small change that I may have missed to mention explicitly.

Reply:
Sounds good!... I think I know what you mean...

Reply:
I had defined the function on top of the file but then I moved it from top of the file to where the tests are that use the function. Please let me know how it looks